### PR TITLE
Feature/silabs doc

### PR DESCRIPTION
**Pull request by mkardous-silabs**
> !!!!!!!!!! Please delete the instructions below and replace with PR description
>
> If you have an issue number, please use a syntax of
> `Fixes #12345` and a brief change description
>
> If you do not have an issue number, please have a good description of
> the problem and the fix. Help the reviewer understand what to expect.
>
> Make sure you delete these instructions (to prove you have read them).
>
> !!!!!!!!!! Instructions end

### CODE REVIEW

File: examples/platform/silabs/README.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+$ ./scripts/examples/gn_silabs_example.sh examples/lighting-app/silabs/ out/lighting-app_wf200 BRD4187C --wifi wf200
+```
+> **Note**: The build argument  `--wifi wf200` is necessary to build the BRD4187C image with the necessary code for the NCP combo.
+
+#### Build Arguments
+
+The ``gn_silabs_examples.sh` script takes two types of build arguments.
+The first type are Macros processed within the script itself and the second are GN arguments.
+The Macros encapsulate multiple multiple GN arguments to simplify enabling or disabled specific features.
+
```
Comment (jmartinez-silabs) [inline examples/platform/silabs/README.md:125]: maybe un petit `./scripts/examples/gn_silabs_examples.sh ` will print a HELP

File: examples/platform/silabs/README.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    -   [Build Script](#Build-Script)
+    -   [Build Arguments](#device-tracing)
+-   [Flashing](#Flashing)
+
+## Requirements
+
+### Hardware Requirements
+
+For the list of hardware requirements, see the official
+[Silicon Labs Matter HW requirements](https://siliconlabs.github.io/matter/2.2.0-1.2/general/HARDWARE_REQUIREMENTS.html)
```
Comment (jmartinez-silabs) [inline examples/platform/silabs/README.md:32]: they link devrait utilisé latest au lieu de la version du release example 
```suggestion
[Silicon Labs Matter HW requirements](https://siliconlabs.github.io/matter/latest/general/HARDWARE_REQUIREMENTS.html)
```

File: examples/platform/silabs/README.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+For pre-built binaries for the latest Silicon Labs Matter release, see the
+official
+[Silicon Labs Matter Software Artifacts](https://siliconlabs.github.io/matter/2.2.0-1.2/general/ARTIFACTS.html#matter-software-artifacts).
+This includes all necessary binairies to run a Silicon Labs sample app.
+
+## Building
+
+Silicon Labs currently supports the following list of sample apps in the main
+Matter SDK. Each Sample has its own documentation to explicit its specificities.
```
Comment (jmartinez-silabs) [inline examples/platform/silabs/README.md:50]: C'est peut-être juste moi mais je trouve ça sonne weird. Maybe this ?
```suggestion
Matter SDK. Every sample has its own documentation explaining its unique features and functionalities.
```

File: examples/platform/silabs/README.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+The examples in the `CSA Matter Repository` column are supported in the main
+Matter SDK. The examples in
+<a href="https://github.com/SiliconLabs/matter">Silabs Matter Repository</a> are
+supported in the Silabs Matter repository.
```
Comment (jmartinez-silabs) [inline examples/platform/silabs/README.md:54]:
```suggestion
Matter SDK.  Additionally, the <a href="https://github.com/SiliconLabs/matter">Silabs Matter Repository</a> offers extra sample applications for different device-types
```

File: examples/platform/silabs/README.md [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+<table>
+    <tbody>
+        <tr>
+            <th>CSA Matter Repository</th>
+            <th> <a href="https://github.com/SiliconLabs/matter">Silabs Matter Repository</a></th>
+        </tr>
+        <tr>
+          <td>
+            <ul>
+                <li> <a href="examples/lighting-app/silabs/README.md">Lighting App</a></li>
```
Comment (mkardous-silabs) [inline examples/platform/silabs/README.md:64]: all paths are wrong

File: docs/guides/silabs_cli_guide.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
+# Silabs CLI Guilde
```
Comment (jmartinez-silabs) [inline docs/guides/silabs_cli_guide.md:1]:
```suggestion
# Silabs CLI Guide
```

File: docs/guides/silabs_cli_guide.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+The Openthread CLI is enabled by default on all Openthread builds. To disable the Openthread CLI, the `enable_openthread_cli=false` argument can be added.
+Here is an example of the build command to disable the Openthread CLI.
+```sh
+./scripts/examples/gn_silabs_example.sh ./examples/lighting-app/silabs/ ./out/lighting-app BRD4187C enable_openthread_cli=false
+```
+
+## Connecting to the device
+
+The different CLI are provided throught the UART interface. The following list explicits the UART configurations to connect to the different CLIs
```
Comment (jmartinez-silabs) [inline docs/guides/silabs_cli_guide.md:38]:
```suggestion
The different CLI are provided through the UART interface. The following table exposes the UART configurations to connect to the different CLIs
```

File: docs/guides/silabs_cli_guide.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+| Configuration | Value | 
+| :----: | :---------- | 
+| Baudrate (speed) | 115200 |
+| Data  | 8 bit |
+| Parity | None |
+| Stop Bit | 1 bit |
+| Flow Control | None |
+
+Any terminal emulator will permit to connect to the device. For MacOS and Linux, the screen utility can be used. For Windows, Terra Term can be used.
```
Comment (jmartinez-silabs) [inline docs/guides/silabs_cli_guide.md:48]: I believe it needs to be a serial terminal

File: docs/guides/silabs_cli_guide.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+### Screen utility
+
+To use the Screen utility, we first need to find the vcom port. For MacOS, it will be formatted as `/dev/cu.usbmodem` and for Linux it will be formatted as `/dev//dev/tty.usbmodem`.
```
Comment (jmartinez-silabs) [inline docs/guides/silabs_cli_guide.md:52]: for Linux it is /dev/ttyACM0 where there 0 can vary to 1, 2 ...

File: docs/guides/silabs_common_app_behavior.md [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+-   [Buttons](#buttons)
+    -   [Opreation Button](#operation-button)
+    -   [Application Button](#application-button)
+-   [LEDs](#leds)
+    -   [Status LED](#status-led)
+    -   [Application LED ](#application-led)
+
+## LCD Screens
+
+When using a developement kit that supports the LCD, the application has three disctint windows.
```
Comment (jmartinez-silabs) [inline docs/guides/silabs_common_app_behavior.md:21]:
```suggestion
When using a development kit that supports the LCD, the application has three distinct windows.
```

File: docs/guides/silabs_getting_started.md [hunk 11]
```diff
@@ -0,0 +1,225 @@
...
+> **Note**: All GN arguments can be added to the build. `gn args --list <output_directory>` can be used to list all GN arguments.
+
+| GN argument | Description | Default Value |
+| :--------:  | :---------- |:----------------- |
+| chip_build_libshell | Enables the Matter Shell | false |
+| chip_openthread_ftd | Defines if the Opentread is an FTD (true) or an MTD (false) | true |
+| efr32_sdk_root | Location for an alternate Gecko SDK | ./third_party/silabs/gecko_sdk |
+| enable_heap_monitoring | Monitor & log memory usage at runtime | false |
+| enable_openthread_cli | Enables the OpenThread cli without the Matter Shell | true |
+| kvs_max_entries | Set the maxium KVS entries that can be stored in NVM <br /> Thresholds: 30 <= kvs_max_entries <= 255 | 75 |
```
Comment (jmartinez-silabs) [inline docs/guides/silabs_getting_started.md:160]: This isn't true anymore, the default is 255. Users can reduce it to save some space. but it is not as impactful as before.

File: docs/guides/silabs_cli_guide.md (hunk 8 above)
Comment (mkardous-silabs) [reply docs/guides/silabs_cli_guide.md:48]: done

File: docs/guides/silabs_cli_guide.md (hunk 9 above)
Comment (mkardous-silabs) [reply docs/guides/silabs_cli_guide.md:52]: done

File: docs/guides/silabs_cli_guide.md [hunk 12]
```diff
@@ -0,0 +1,195 @@
+# Silabs CLI Guide
+
+## Introduction
+The guide discusses the different CLIs that can be enabled with the Silabs sample apps.
+The CLIs expose configuration and management APIs via a command line interface (CLI).
+For Openthread devices, the Openthread CLI can be used with or without the Matter CLI. For Wi-Fi devices, only the Matter CLI is available.
```
Comment (jrsilabs) [inline docs/guides/silabs_cli_guide.md:6]: OpenThread (capital T) in both instances

File: docs/guides/silabs_common_app_behavior.md [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+## LEDs
+
+All sample applications are designed to work with two LEDs : the application LED and the status LED.
+LED0 is the status LED and the LED1 is the application LED.
+>**Note:** Some dev kits can only support the buttons or the LEDs. The button support is the default configuration. <br/>
+> For dev kits with only LED, the application LED is the default configuration.
+
+### Status LED
+
+The following list explicits all the states of the status LED.
```
Comment (jrsilabs) [inline docs/guides/silabs_common_app_behavior.md:109]: describes instead of explicits

File: docs/guides/silabs_getting_started.md [hunk 14]
```diff
@@ -0,0 +1,225 @@
...
+
+![Silabs logo](./images/silabs_logo.png)
+
+> **NOTE:** Silicon Laboratories now maintains a public Matter GitHub repo with
+> frequent releases thoroughly tested and validated. Developers looking to
+> develop matter products with silabs hardware are encouraged to use our latest
+> release with added tools and documentation.
+> [Silabs Matter Github](https://github.com/SiliconLabs/matter/releases)
+>
+> Developers can find more resources on
```
Comment (jrsilabs) [inline docs/guides/silabs_getting_started.md:17]: ...on the...

File: docs/guides/silabs_getting_started.md [hunk 15]
```diff
@@ -0,0 +1,225 @@
...
+> [Silicon Labs Matter Community Page](https://community.silabs.com/s/article/connected-home-over-ip-chip-faq?language=en_US).
+
+
+-   [Introduction](#introduction)
+-   [Requirements](#requirements)
+    -   [Hardware Requirements](#hardware-requirements)
+    -   [Software Requirements](#software-requirements)
+    -   [Software Artificats](#software-artificats)
```
Comment (jrsilabs) [inline docs/guides/silabs_getting_started.md:25]: spelling: Artifacts
should change the link text and the anchor

File: docs/guides/silabs_getting_started.md [hunk 16]
```diff
@@ -0,0 +1,225 @@
...
+
+For the list of software requirements, see the official
+[Silicon Labs Matter Software requirements](https://siliconlabs.github.io/matter/latest/general/SOFTWARE_REQUIREMENTS.html)documentation.
+
+#### Software Artificats
+
+For pre-built binaries for the latest Silicon Labs Matter release, see the
+official
+[Silicon Labs Matter Software Artifacts](https://siliconlabs.github.io/matter/latest/general/ARTIFACTS.html#matter-software-artifacts).
+This includes all necessary binairies to run a Silicon Labs sample app.
```
Comment (jrsilabs) [inline docs/guides/silabs_getting_started.md:51]: spelling: binaries

File: docs/guides/silabs_getting_started.md [hunk 17]
```diff
@@ -0,0 +1,225 @@
...
+
+## Building
+
+Silicon Labs currently supports the following list of sample apps in the main
+Matter SDK. Every sample has its own documentation explaining its unique features and functionalities.
+The examples in the `CSA Matter Repository` column are supported in the main Matter SDK.
+Additionally, the <a href="https://github.com/SiliconLabs/matter">Silabs Matter Repository</a> offers extra sample applications for different device-types
```
Comment (jrsilabs) [inline docs/guides/silabs_getting_started.md:58]: will this work? Shouldn't the link be [Silabs Matter Repository](https://github.com/SiliconLabs/matter)?

File: docs/guides/silabs_getting_started.md [hunk 18]
```diff
@@ -0,0 +1,225 @@
...
+        </tr>
+    </tbody>
+</table>
+
+
+### Build Script
+
+To build a Silicon Sample apps, we provide the `gn_silabs_examples.sh` scripts that can be found in the `./scripts/examples` directory.
+The build script can be used to build all of the Silabs supported examples.
+The command structure is as follows when called from the root of the repository :
```
Comment (jrsilabs) [inline docs/guides/silabs_getting_started.md:94]: remove space before the colon

File: docs/guides/silabs_getting_started.md [hunk 19]
```diff
@@ -0,0 +1,225 @@
...
+To build the lighting app as an Openthread SoC, the default build command for the BRD4187C DK is
+```shell
+./scripts/examples/gn_silabs_example.sh ./examples/lighting-app/silabs/ ./out/lighting-app BRD4187C
+```
+
+To build the lighting app as an Wi-Fi MG24 + RS9116 NCP, the default build command for the BRD4187C is
+```shell
+./scripts/examples/gn_silabs_example.sh examples/lighting-app/silabs/ out/lighting-app_rs9116 BRD41xxx use_external_flash=false chip_enable_ble_rs911x=true --wifi rs9116
+```
+> **Note**: The build argument  `--wifi rs9116` is necessary to build BRD4187C image with the necessary code for the NCP combo.
```
Comment (jrsilabs) [inline docs/guides/silabs_getting_started.md:108]: delete extra space after "argument"

File: docs/guides/silabs_getting_started.md [hunk 20]
```diff
@@ -0,0 +1,225 @@
...
+```
+> **Note**: The build argument  `--wifi wf200` is necessary to build the BRD4187C image with the necessary code for the NCP combo.
+
+#### Build Arguments
+
+The ``gn_silabs_examples.sh` script takes two types of build arguments.
+The first type are Macros processed within the script itself and the second are GN arguments.
+The Macros encapsulate multiple multiple GN arguments to simplify enabling or disabled specific features.
+
+> **Note**: Executing the build script without any arguments will print a helper with the command structure, the list of supported boards and a list of supported macros and arguments
```
Comment (jrsilabs) [inline docs/guides/silabs_getting_started.md:130]: add a comma after ...supported boards, ...

File: docs/guides/silabs_getting_started.md [hunk 21]
```diff
@@ -0,0 +1,225 @@
...
+> ```sh
+> ./scripts/examples/gn_silabs_examples.sh
+> ```
+
+Here is a list of some the supported macros and their GN argument equivalent.
+| Macro Name | Description                     | GN equivalent                                         |
+| :-------:  | :----------                     | :---                                                  |
+| --wifi | Configures an sample app as a Wi-Fi devices.<br /> This macro requires rs9116 or SiWx917 or wf200. | --wifi rs9116 : use_rs9116=true<br /> --wifi SiWx917 : use_SiWx917=true<br /> --wifi wf200 : use_wf200=true<br /> |
+| --icd | Configures the device as an ICD | chip_enable_icd_server=true chip_openthread_ftd=false |
+| --low-power | Configures the most power efficient build.<br /> This is used in tadem with the `--icd` macro | chip_build_libshell=false enable_openthread_cli=false show_qr_code=false disable_lcd=true |
```
Comment (jrsilabs) [inline docs/guides/silabs_getting_started.md:140]: spelling: tandem, instead of tadem

File: docs/guides/silabs_getting_started.md [hunk 22]
```diff
@@ -0,0 +1,225 @@
...
+| --chip_enable_wifi_ipv4 | Enables IPv4 support on Wi-fi configured builds | chip_enable_wifi_ipv4=true chip_inet_config_enable_ipv4=true |
+| --clean | Cleans the output directory before building | NA |
+| --additional_data_advertising | Enable Addition data advertissing and rotating device ID | chip_enable_additional_data_advertising=true chip_enable_rotating_device_id=true |
```
Comment (jrsilabs) [inline docs/guides/silabs_getting_started.md:143]: spelling: advertising

File: docs/guides/silabs_getting_started.md [hunk 23]
```diff
@@ -0,0 +1,225 @@
...
+| --use_ot_lib | Builds the sample app with the Silabs certified Openthread librairies | use_silabs_thread_lib=true chip_openthread_target=$SILABS_THREAD_TARGET openthread_external_platform=\"""\" |
```
Comment (jrsilabs) [inline docs/guides/silabs_getting_started.md:144]: spelling: libraries

File: docs/guides/silabs_getting_started.md [hunk 24]
```diff
@@ -0,0 +1,225 @@
...
+| --use_ot_coap_lib | Builds the sample app with the Silabs certified Openthread COAP librairies | use_silabs_thread_lib=true chip_openthread_target=$SILABS_THREAD_TARGET openthread_external_platform=\"""\" use_thread_coap_lib=true |
```
Comment (jrsilabs) [inline docs/guides/silabs_getting_started.md:145]: spelling: libraries

File: docs/guides/silabs_getting_started.md [hunk 25]
```diff
@@ -0,0 +1,225 @@
...
+| --release| Remove all logs and debugs features (including the LCD). <br />Yields the smallest image size possible | is_debug=false disable_lcd=true chip_build_libshell=false enable_openthread_cli=false use_external_flash=false chip_logging=false silabs_log_enabled=false |
+| --bootloader | Adds a bootloader to the built image | NA |
+| --uart_log | Forwards logs to uart instead of RTT | sl_uart_log_output=true |
+
+Here is a list of some of the GN arguments that can be added the build command.
```
Comment (jrsilabs) [inline docs/guides/silabs_getting_started.md:150]: ...that can be added to the build command.

File: docs/guides/silabs_getting_started.md [hunk 26]
```diff
@@ -0,0 +1,225 @@
...
+> **Note**: All GN arguments can be added to the build. `gn args --list <output_directory>` can be used to list all GN arguments.
+
+| GN argument | Description | Default Value |
+| :--------:  | :---------- |:----------------- |
+| chip_build_libshell | Enables the Matter Shell | false |
+| chip_openthread_ftd | Defines if the Opentread is an FTD (true) or an MTD (false) | true |
+| efr32_sdk_root | Location for an alternate Gecko SDK | ./third_party/silabs/gecko_sdk |
+| enable_heap_monitoring | Monitor & log memory usage at runtime | false |
+| enable_openthread_cli | Enables the OpenThread cli without the Matter Shell | true |
+| kvs_max_entries | Set the maxium KVS entries that can be stored in NVM <br /> Thresholds: 30 <= kvs_max_entries <= 255 | 255 |
```
Comment (jrsilabs) [inline docs/guides/silabs_getting_started.md:160]: spelling: maximum

File: docs/guides/silabs_getting_started.md [hunk 27]
```diff
@@ -0,0 +1,225 @@
...
+```
+To execute the script, the following command can be used:
+```sh
+python3 <path>/matter-silabs-lighting-example.flash.py
+```
+where `<path>` is the path to the output directory used in the build script.
+
+> **Note**: It is also possible to flash the built binary with commander directly.
+> The commander command is `commander flash  <path>/matter-silabs-lighting-example.<s37/rps>` <br />
+> The `.s37` binaires are used with the MGM24 and EFR32 families while the `.rps` binaires are only used for the SiWx917 SoC family.
```
Comment (jrsilabs) [inline docs/guides/silabs_getting_started.md:187]: spelling: binaries; two instances

File: docs/guides/silabs_getting_started.md [hunk 28]
```diff
@@ -0,0 +1,225 @@
...
+
+
+
+### Flasher Arguments
+
+The flashing scripts provides configuration arguments and operation arguments.
```
Comment (jrsilabs) [inline docs/guides/silabs_getting_started.md:193]: spelling: provide (singular)

File: docs/guides/silabs_getting_started.md [hunk 29]
```diff
@@ -0,0 +1,225 @@
...
+| --commander FILE | Path to the commander executable |
+| --device, -d DEVICE | Device family or platform to target (EFR32, MGM24, 917) - TODO |
+| --serialno, -s SERIAL | Serial number of device to flash. <br /> This argument is necessary when multiple boards of the same family are connected. |
+| --ip, -a  ADDRESS | IP Address of the device to flash |
+
+Here is a list of the operations arguments that can be added to the flasher script.
+
+| Argument | Description | Commander equivalent |
+| :------: | :-----------| :--------------------|
+| --erase | Erase the devices flash. <br /> This options completly wipes the devices flash including factory data. | `commander device masserase`
```
Comment (jrsilabs) [inline docs/guides/silabs_getting_started.md:210]: spelling: completely

File: docs/guides/silabs_getting_started.md (hunk 17 above)
Comment (mkardous-silabs) [reply docs/guides/silabs_getting_started.md:58]: both work, don't remember why i used the html version

**Pull request closed without merging**